clang/lib/AST/ASTContext.cpp

```diff
@@ -3140,13 +3184,20 @@ const FunctionType *ASTContext::adjustFunctionType(const FunctionType *T,
   return cast<FunctionType>(Result.getTypePtr());
 }
 
+QualType ASTContext::adjustFunctionResultType(QualType FunctionType,
+                                              QualType ResultType) {
+  return adjustType(FunctionType, [&](QualType Orig) {
+    const auto *FPT = Orig->castAs<FunctionProtoType>();
```

Comment:
There's a bit of a gotcha here -- functions without prototypes have a result type that could be adjusted, so this will cause an assertion that the caller is unlikely to expect. Can we have a fallback for functions without prototypes and some test coverage for those?

Reply:
Ah, good point. I forgot that prototypeless functions can still have result types.

(Side note: I also just somehow managed to edit your comment instead of posting this one...)

Reply:
This now handles `FunctionNoProtoType`s as well.